libcontainer/rootfs_linux.go

```diff
@@ -572,6 +572,11 @@ func pivotRoot(rootfs, pivotBaseDir string) error {
 	if err != nil {
 		return fmt.Errorf("can't create pivot_root dir %s, error %v", pivotDir, err)
 	}
+	defer func() {
+		if errVal := os.Remove(pivotDir); errVal != nil && err == nil {
+			err = errVal
```

Comment:
If the err is not nil.. we loose the additional msg that the pivot directory was created but could not be removed... can that be concatenated on the end of the err being returned?  Additionally the os.Remove error isn't very explanatory can we add descriptive text here?